packages/embed/src/index.ts

```diff
@@ -18,11 +29,20 @@ import { validate } from './validation'
  */
 export function setup(setupConfig: SetupConfig): void {
   const { gr4vyId, onComplete, ...rest } = setupConfig
+
+  const channel = generateChannelId()
+  const iframeHost = gr4vyId
+    ? `embed.${gr4vyId}.gr4vy.app`
+    : setupConfig.iframeHost
+  const apiHost = gr4vyId ? `api.${gr4vyId}.gr4vy.app` : setupConfig.apiHost
+
   const config: Config = {
```

Comment:
This is the confusion of `Config` - We pass in `SetupConfig`, ideally we should validate that and then create a separate Config - I think this could be moved out into a `createConfig` function that takes a `SetupConfig` so we can fully test it.

Reply:
I made this change to allow me to test building the config more easily.